examples/serial_test.py

```diff
@@ -0,0 +1,28 @@
+from robocluster import Device
+
+
+device = Device('link', 'rover')
+device.create_serial('/dev/ttyACM0')
```

Comment:
Would `serial = device.serial('/dev/ttyACM0')` followed by `@serial.on('test')` be better syntax?

It would allow for:
```py
motor_a = device.serial('/path/to/motor_a/tty')
motor_b = device.serial('/path/to/motor_b/tty')

@motor_a.on('some_event')
async def callback(event, data):
    pass
```

Reply:
Additionally we could support, this is probably something for a different pull request...
```py
async def motor_callback(event, data):
    pass
motor_a.on('some_event', motor_callback)
motor_b.on('some_event', motor_callback)
```

Reply:
I just added this, but one issue is regarding using this at the same time as manually reading from the serial device. I'm not sure how to handle the case of having a read loop for the callbacks calling `SerialDevice.read_packet()`, and other tasks potentially calling `SerialDevice.read_byte()`. Could the read loop put the bytes of packets that do not contain events it is subscribed to in a separate queue, and the read_byte() methods read from this queue instead? It would give greater flexibility, but responsiveness might be poor and the logic could be more prone to bugginess. The alternative is to disallow calls to SerialDevice read methods if callbacks are registerd with `serial.on('event')`.